sklearn/utils/validation.py

```diff
@@ -363,8 +363,11 @@ def check_array(array, accept_sparse=None, dtype="numeric", order=None,
     else:
         if ensure_2d:
             array = np.atleast_2d(array)
-
-        array = np.array(array, dtype=dtype, order=order, copy=copy)
+        # If copy=False and array is a np.memmap, we avoid copying array by using function np.asarray
+        if not copy:
+            array = np.asarray(array, dtype=dtype, order=order)
+        else:
+            array = np.array(array, dtype=dtype, order=order, copy=copy)
```

Comment:
I wonder if instead we should not do:

1- always use `np.asarray` here.
2- set `array_orig = array` at the beginning of the function
3- then at the end of this function, just before returning:

```python
if copy:
    array = array.copy()
```